ts/stateTree.ts

```diff
@@ -92,10 +93,10 @@ export class StateTree implements StateProvider {
     public static new(stateDepth: number) {
         return new StateTree(stateDepth);
     }
-    private stateTree: Tree;
+    private stateTree: MemoryTree;
```

Comment:
Unsure if this should be a memory or database tree, @ChihChengLiang can you provide insight?

Reply:
As all classes in the stateTree.ts here are currently used in the tests, using MemoryTree should be fine.

Eventually, we want to replace all classes in the stateTree.ts with the classes used by the client.